### PR TITLE
[PREVIEW] Search for bar payments

### PR DESCRIPTION
**Pull request by tarun-palisetty**

### CODE REVIEW

File: model/src/main/java/uk/gov/hmcts/payment/api/util/PaymentMethodType.java [hunk 1]
```diff
@@ -3,7 +3,10 @@
 public enum PaymentMethodType {
 
     CARD("card"),
-    PBA("payment by account");
+    PBA("payment by account"),
+    CASH("cash"),
+    CHEQUE("cheque"),
```
Comment (smathangi) [inline model/src/main/java/uk/gov/hmcts/payment/api/util/PaymentMethodType.java]: We've a similar enum Method.java . Do you mind to remove this and update references with  PaymentMethodType if possible? 
https://github.com/hmcts/ccpay-payment-app/blob/master/api-contract/src/main/java/uk/gov/hmcts/payment/api/contract/util/Method.java

Comment (tarun-palisetty) [reply model/src/main/java/uk/gov/hmcts/payment/api/util/PaymentMethodType.java]: The existing /payments implementation uses PaymentMethodType enum

paymentMethodType.map(value -> PaymentMethodType.valueOf(value.toUpperCase()).getType()).orElse(null);

So I introduced the new types here.

Comment (smathangi) [reply model/src/main/java/uk/gov/hmcts/payment/api/util/PaymentMethodType.java]: Yes, that's right.  But I'm asking a cleanup - delete duplicate enum (Method.java) if possible.

Comment (tarun-palisetty) [reply model/src/main/java/uk/gov/hmcts/payment/api/util/PaymentMethodType.java]: cheers, will do it

File: api/src/main/java/uk/gov/hmcts/payment/api/dto/mapper/CardPaymentDtoMapper.java [hunk 2]
```diff
@@ -95,6 +95,9 @@ public PaymentDto toReconciliationResponseDto(PaymentFeeLink paymentFeeLink) {
             .dateCreated(payment.getDateCreated())
             .dateUpdated(payment.getDateUpdated())
             .method(payment.getPaymentMethod().getName())
+            .giroSlipNo(payment.getGiroSlipNo())
```
Comment (jaleen) [inline api/src/main/java/uk/gov/hmcts/payment/api/dto/mapper/CardPaymentDtoMapper.java]: pls generalise DTO name

**Pull request merged**